internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -177,7 +177,13 @@ func (s Service) ListAuthMethods(ctx context.Context, req *pbs.ListAuthMethodsRe
 			return nil, err
 		}
 
-		if filter.Match(item) {
+		// This comes last so that we can use item fields in the filter after
+		// the allowed fields are populated above
+		filterable, err := subtypes.Filterable(item)
+		if err != nil {
+			return nil, err
+		}
+		if filter.Match(filterable) {
```

Comment:
Forgot to add the filterable conversion here, and it turns out we didn't have any tests for this.

Reply:
hmm yea, good catch, yea some tests for this sounds like a good idea

Reply:
Addressed in fce9968e68ae923c71f0a51500590f732e52ef5e